shipment_advice/models/shipment_advice.py

```diff
@@ -238,7 +238,7 @@ def _compute_carrier_ids(self):
     def create(self, vals):
         defaults = self.default_get(["name", "shipment_type"])
         sequence = self.env.ref("shipment_advice.shipment_advice_outgoing_sequence")
-        if defaults["shipment_type"] == "incoming":
+        if vals["shipment_type"] == "incoming":
```

Comment:
Then there is no need to request `shipment_type` on `default_get()` call above as it is not used anymore.